import "goby/protobuf/option_extensions.proto";

package goby.transitional.protobuf;

message DCCLTransitionalConfig
{
  optional int32 modem_id = 1 [(description)="Unique number 1-31 to identify this node", (example)="1"];
  repeated MessageFile message_file = 2 [(description)="XML message file containing one or more DCCL message descriptions"];
  
  optional string generated_proto_dir = 4 [(description)="Directory to store auto-generated .proto files from deprecated XML files", default="/tmp"];
}


message MessageFile
{
  required string path = 1 [(description)="path to the message XML file", (example)="/home/toby/goby/src/acomms/examples/chat/chat.xml"];
  enum Manipulator 
  {
    NO_MANIP = 1; // blank "empty" manipulator
    
    // libdccl (DCCLCodec)
    NO_ENCODE = 2;
    NO_DECODE = 3;

    // libqueue (QueueManager)
    NO_QUEUE = 4;
    LOOPBACK = 5;
  }
  repeated Manipulator manipulator = 2 [(description)="manipulators to modify the encoding and queuing behavior of the messages in this file", (example)="NO_MANIP"];
}
